### DISCUSSION TITLE
更适合看经济实证论文的笔记模板

### DISCUSSION THREAD

**Discussion by hzp2333** [windingwind/zotero-better-notes#973] [Note Templates]
### Description

更适合看经济实证论文的笔记模板
调整自 #771 
需要translate、zotero style插件支持
版本:
`Zotero 7.0.0-beta.77+adaa61f2c版本`+`Ethereal Style4.4.2版本`+`Better note1.1.4-49`。

> 现在换成了[zotero 7稳定版](https://www.zotero.org/download/)依旧可以用

### Screenshots

![8d61fe34ba1ada661e257491aae40f9](https://github.com/windingwind/zotero-better-notes/assets/79459902/7c658ab0-36ce-493c-8166-e5b49fe43ce1)

![sdfsdf](https://github.com/user-attachments/assets/55e67c8b-b32b-4560-a160-1e7ca999d61b)

### Test on Zotero version

Zotero 7.0.0-beta.77+adaa61f2c
> 现在换成了[zotero 7稳定版](https://www.zotero.org/download/)依旧可以用

### Test on Better Notes version

1.1.4-49

### Template Type

Item

### Template Share Code

```<!-- 标题 -->
  <h1 style="color:#193c47; background-color:#eef9fd; padding:8px;">
    ${(() => {
      const titleTranslation = topItem.getField("titleTranslation");
      if (titleTranslation) {
        return `(${topItem.getField("date")}) ${topItem.getField("title")} (${titleTranslation})`;
      } else {
        return `(${topItem.getField("date")}) ${topItem.getField("title")}`;
      }
    })()}
  </h1>
  
  <!--元数据-->
  <table>
    <!-- 作者 -->
    <tr>
      <td style="color:#193c47; background-color:#dbeedd; padding:8px;">
        <b>作者:</b> ${topItem.getCreators().map((v) => v.firstName + " " + v.lastName).join("; ")}
      </td>
    </tr>
  
    <!-- 期刊 -->
    <tr>
      <td style="color:#193c47; background-color:#f3faf4; padding:8px;">
        <b style="color:#193c47;">期刊: <b style="color:#FF0000">${topItem.getField('publicationTitle')}</b></b><b style="color:#193c47;"> （发表日期: ${topItem.getField("date")}）</b>
      </td>
    </tr>
  
    <!-- 期刊分区 -->
    <tr>
      <td style="color:#193c47; background-color:#dbeedd; padding:8px;">
        <b>期刊分区: </b>
        <!-- Zotero7中，引用了Ethereal Style插件的标签，请提前安装Ethereal Style-->
        ${{
        let space = " ㅤㅤ ㅤㅤ"
        return Array.prototype.map.call(
          Zotero.ZoteroStyle.api.renderCell(topItem, "publicationTags").childNodes,
          e => {
            e.innerText =  space + e.innerText + space;
            return e.outerHTML
          }
          ).join(space)
        }}$
      </td>
    </tr>

    <!-- 本地链接 -->
    <tr>
      <td style="color:#193c47; background-color:#f3faf4; padding:8px;">
        ${(() => {
          const attachments = Zotero.Items.get(topItem.getAttachments());
          if (attachments && attachments.length > 0) {
            return `<b>本地链接: </b><a href="zotero://open-pdf/0_${attachments[0].key}">${attachments[0].getFilename()}</a>`;
          } else {
            return `<b>本地链接: </b>`;
          }
        })()}
      </td>
    </tr>

    <!-- DOI or URL -->
    <tr>
      <td style="color:#193c47; background-color:#dbeedd; padding:8px;">
        ${(() => {
          const doi = topItem.getField("DOI");
          if (doi) {
            return `<b>DOI: </b><a href="https://doi.org/${topItem.getField('DOI')}">${topItem.getField('DOI')}</a>`;
          } else {
            return `<b>URL: </b><a href="${topItem.getField('url')}">${topItem.getField('url')}</a>`;
          }
        })()}
      </td>
    </tr>
    
    <!-- 摘要 -->
    <tr>
      <td style="color:#193c47; background-color:#f3faf4; padding:8px;">
        ${(() => {
          const abstractTranslation = topItem.getField('abstractTranslation');
          if (abstractTranslation) {
            return `<b>摘要翻译: </b><i>${abstractTranslation}</i>`;
          } else {
            return `<b>摘要: </b><i>${topItem.getField('abstractNote')}</i>`;
          }
        })()}
      </td>
    </tr>
  
    <!-- 笔记日期 -->
    <tr>
      <td style="color:#193c47; background-color:#dbeedd; padding:8px;">
        <b>笔记日期: </b>${new Date().toLocaleString()}
      </td>
    </tr>
  
  </table>
  
  <!-- 正文 -->
  <span>
    <h2 style="color:#e0ffff; background-color:#66cdaa;">📜 研究核心</h2>
    <hr />
  </span>
  <blockquote>Tips: 做了什么，解决了什么问题，创新点与不足？</blockquote>
  以xx为对象，使用了xx方法，得到了xx结论
  <p></p>
 
  <span>
    <h2 style="color:#20b2aa; background-color:#afeeee;">🔁 实证部分</h2>
    <hr />
  </span>
  <p></p>
  <h3>💧 模型</h3>
  <blockquote>Tips: 理论现象路径背景和计量回归式子</blockquote>
  正文开始
  <p></p>
  <h3>👩🏻‍💻 数据</h3>
  <table border="1">
  <tr>
    <td style="color:#193c47; background-color:#dbeedd; padding:8px;"><b>变量</b></td>
    <td style="color:#193c47; background-color:#dbeedd; padding:8px;"><b>含义</b></td>
    <td style="color:#193c47; background-color:#dbeedd; padding:8px;"><b>数据源</b></td>
  </tr>
  <tr>
    <td>因变量</td>
    <td> </td>
  </tr>
  <tr>
    <td>自变量</td>
    <td> </td>
    <td> </td>
  </tr>
  <tr>
    <td>控制变量</td>
    <td> </td>
    <td> </td>
  </tr>

    <tr>
    <td>其他变量IV等</td>
    <td></td>
    <td></td>
  </tr>

      <tr>
    <td>效应</td>
    <td>双向固定、聚类</td>
    <td></td>
  </tr>
  
</table>

  <p></p>
  <h3>🔬 稳健性与内生性</h3>
  <p></p>
  <h3>📜 异质性</h3>
  <p></p>

	<h3>⚙️ 机制检验</h3>
  <p></p>

  <h3>💡 拓展分析</h3>
  <p></p>
  
  <span>
    <h2 style="color:#004d99; background-color:#87cefa;">🤔 总结</h2>
    <hr />
  </span>
  <blockquote>Tips: 你对哪些内容产生了疑问，你认为可以如何改进？</blockquote>
  <p></p>
  <h3>🙋‍♀️ 文章结论</h3>
  <blockquote>Tips: 实然：因果关系结论+应然：政策启示</blockquote>
  <p></p>
  <h3>📌 方法启示 </h3>
  <blockquote>Tips: 怎么设计的实验？怎么构造的函数？怎么演绎经济故事逻辑？</blockquote>
  <p></p>
  <h3>💭 思考启发</h3>
  <p></p>
```

**Comment by Carriekk** [comment]
作者您好！请问有适配zotero 6版本的嘛？

**Reply by hzp2333** [reply]
zotero6和zotero7模板差距主要是zotero style插件读取文件元数据。因此理论上只需要修改zotero style读取部分。
你可以试试看这个。
```
<!-- 标题 -->
  <h1 style="color:#193c47; background-color:#eef9fd; padding:8px;">
  ${(() => {
    const titleTranslation = topItem.getField("titleTranslation");
    if (titleTranslation) {
      return `(${topItem.getField("date")}) ${topItem.getField("title")} (${titleTranslation})`;
    } else {
      return `(${topItem.getField("date")}) ${topItem.getField("title")}`;
    }
  })()}
</h1>
  
  <!-- Meta Data -->
  <table>
  <!-- 作者 -->
  <tr>
  <td style="color:#193c47; background-color:#dbeedd; padding:8px;">
  <b>作者:</b> ${topItem.getCreators().map((v) => v.firstName + " " + v.lastName).join("; ")}
</td>
  </tr>
  
  <!-- 期刊 -->
  <tr>
  <td style="color:#193c47; background-color:#f3faf4; padding:8px;">
  <b style="color:#193c47;">期刊: <b style="color:#FF0000">${topItem.getField('publicationTitle')}</b></b><b style="color:#193c47;"> （发表日期: ${topItem.getField("date")}）</b>
  </td>
  </tr>
  
  <!-- 期刊分区 -->
  <tr>
  <td style="color:#193c47; background-color:#dbeedd; padding:8px;">
  <b>期刊分区: </b>
  <!-- 引用了zotero style插件的标签，请提前安装，否则找不到分区 -->
  ${(() => {let space = " ㅤㅤ ㅤㅤ"
  return Array.prototype.map.call(Zotero.ZoteroStyle.data.ztoolkit.ItemTree.globalCache.renderCellHooks.PublicationTags(
    0,
    Zotero.ZoteroStyle.data.ztoolkit.ItemTree.fieldHooks.globalCache.getFieldHooks.PublicationTags(
      "", true, true,
      topItem, undefined)
  ).childNodes,
  e => {
    e.innerText = space + space + e.innerText + space + space;
    return e.outerHTML
  }).join(space)
  })()
  }
</td>
  </tr>
  
  <!-- 本地链接 -->
  <tr>
  <td style="color:#193c47; background-color:#f3faf4; padding:8px;">
  ${(() => {
    const attachments = Zotero.Items.get(topItem.getAttachments());
    if (attachments && attachments.length > 0) {
      return `<b>本地链接: </b><a href="zotero://open-pdf/0_${attachments[0].key}">${attachments[0].getFilename()}</a>`;
    } else {
      return `<b>本地链接: </b>`;
    }
  })()}
</td>
  </tr>
  
  <!-- DOI or URL -->
  <tr>
  <td style="color:#193c47; background-color:#dbeedd; padding:8px;">
  ${(() => {
    const doi = topItem.getField("DOI");
    if (doi) {
      return `<b>DOI: </b><a href="https://doi.org/${topItem.getField('DOI')}">${topItem.getField('DOI')}</a>`;
    } else {
      return `<b>URL: </b><a href="${topItem.getField('url')}">${topItem.getField('url')}</a>`;
    }
  })()}
</td>
  </tr>
  
  <!-- 摘要 -->
  <tr>
  <td style="color:#193c47; background-color:#f3faf4; padding:8px;">
  ${(() => {
    const abstractTranslation = topItem.getField('abstractTranslation');
    if (abstractTranslation) {
      return `<b>摘要翻译: </b><i>${abstractTranslation}</i>`;
    } else {
      return `<b>摘要: </b><i>${topItem.getField('abstractNote')}</i>`;
    }
  })()}
</td>
  </tr>
  
  <!-- 笔记日期 -->
  <tr>
  <td style="color:#193c47; background-color:#dbeedd; padding:8px;">
  <b>笔记日期: </b>${new Date().toLocaleString()}
</td>
  </tr>
  
  </table>
  
  <!-- 正文 -->
  <span>
  <h2 style="color:#e0ffff; background-color:#66cdaa;">📜 研究核心</h2>
  <hr />
  </span>
  <blockquote>Tips: 做了什么，解决了什么问题，创新点与不足？</blockquote>
  以xx为对象，使用了xx方法，得到了xx结论
<p></p>
  
  <span>
  <h2 style="color:#20b2aa; background-color:#afeeee;">🔁 实证部分</h2>
  <hr />
  </span>
  <p></p>
  <h3>💧 模型</h3>
  <blockquote>Tips: 理论现象路径背景和计量回归式子</blockquote>
  正文开始
<p></p>
  <h3>👩🏻‍💻 数据</h3>
  <table border="1">
  <tr>
  <td style="color:#193c47; background-color:#dbeedd; padding:8px;"><b>变量</b></td>
  <td style="color:#193c47; background-color:#dbeedd; padding:8px;"><b>含义</b></td>
  <td style="color:#193c47; background-color:#dbeedd; padding:8px;"><b>数据源</b></td>
  </tr>
  <tr>
  <td>因变量</td>
  <td> </td>
  </tr>
  <tr>
  <td>自变量</td>
  <td> </td>
  <td> </td>
  </tr>
  <tr>
  <td>控制变量</td>
  <td> </td>
  <td> </td>
  </tr>
  
  <tr>
  <td>其他变量IV等</td>
  <td></td>
  <td></td>
  </tr>
  
  <tr>
  <td>效应</td>
  <td>双向固定、聚类</td>
  <td></td>
  </tr>
  
  </table>
  
  <p></p>
  <h3>🔬 稳健性与内生性</h3>
  <p></p>
  <h3>📜 异质性</h3>
  <p></p>
  
  <h3>⚙️ 机制检验</h3>
  <p></p>
  
  <h3>💡 拓展分析</h3>
  <p></p>
  
  <span>
  <h2 style="color:#004d99; background-color:#87cefa;">🤔 总结</h2>
  <hr />
  </span>
  <blockquote>Tips: 你对哪些内容产生了疑问，你认为可以如何改进？</blockquote>
  <p></p>
  <h3>🙋‍♀️ 文章结论</h3>
  <blockquote>Tips: 实然：因果关系结论+应然：政策启示</blockquote>
  <p></p>
  <h3>📌 方法启示 </h3>
  <blockquote>Tips: 怎么设计的实验？怎么构造的函数？怎么演绎经济故事逻辑？</blockquote>
  <p></p>
  <h3>💭 思考启发</h3>
  <p></p>
```

**Reply by szh15839163898** [reply]
作者您好，两个都试了一下
第一次是报错是SyntaxError: missing : after property id 
第二次显示的报错是TypeError: Zotero.ZoteroStyle.data is undefined 
请问改怎么修改呀

**Reply by hzp2333** [reply]
看样子你的是zotero6,第二个基本调用成功了，似乎是zoterostyle插件没安装好？或者说你使用笔记模板没有文章元数据

**Reply by iampearl918** [reply]
<img width="255" alt="截屏2024-08-09 17 57 06" src="https://github.com/user-attachments/assets/c1aa3208-83de-4420-9a84-9d1ee2eb543e">


我复制这个到剪切板时就不行，报错：invalid template。

条目的元数据都没问题的，因为还试用了其他类似模版都正常。

怎么解决呢？

**Reply by hzp2333** [reply]
刚刚下载了zotero6版本实验了下，应该是没有问题的。
请再检查下：
笔记名称是否为`[item]自定义名字`
zotero下载了zotero style插件。

还有个解决方法是，如果其他能导入元数据的模板能正常运行，
那干脆在他们基础上修改中间的正文标题部分就行了。

也就是替换模板代码中论文元数据以外的正文部分，如下：

 ```
<!-- 正文 -->
  <span>
    <h2 style="color:#e0ffff; background-color:#66cdaa;">📜 研究核心</h2>
    <hr />
  </span>
  <blockquote>Tips: 做了什么，解决了什么问题，创新点与不足？</blockquote>
  以xx为对象，使用了xx方法，得到了xx结论
  <p></p>
 
  <span>
    <h2 style="color:#20b2aa; background-color:#afeeee;">🔁 实证部分</h2>
    <hr />
  </span>
  <p></p>
  <h3>💧 模型</h3>
  <blockquote>Tips: 理论现象路径背景和计量回归式子</blockquote>
  正文开始
  <p></p>
  <h3>👩🏻‍💻 数据</h3>
  <table border="1">
  <tr>
    <td style="color:#193c47; background-color:#dbeedd; padding:8px;"><b>变量</b></td>
    <td style="color:#193c47; background-color:#dbeedd; padding:8px;"><b>含义</b></td>
    <td style="color:#193c47; background-color:#dbeedd; padding:8px;"><b>数据源</b></td>
  </tr>
  <tr>
    <td>因变量</td>
    <td> </td>
  </tr>
  <tr>
    <td>自变量</td>
    <td> </td>
    <td> </td>
  </tr>
  <tr>
    <td>控制变量</td>
    <td> </td>
    <td> </td>
  </tr>

    <tr>
    <td>其他变量IV等</td>
    <td></td>
    <td></td>
  </tr>

      <tr>
    <td>效应</td>
    <td>双向固定、聚类</td>
    <td></td>
  </tr>
  
</table>

  <p></p>
  <h3>🔬 稳健性与内生性</h3>
  <p></p>
  <h3>📜 异质性</h3>
  <p></p>

	<h3>⚙️ 机制检验</h3>
  <p></p>

  <h3>💡 拓展分析</h3>
  <p></p>
  
  <span>
    <h2 style="color:#004d99; background-color:#87cefa;">🤔 总结</h2>
    <hr />
  </span>
  <blockquote>Tips: 你对哪些内容产生了疑问，你认为可以如何改进？</blockquote>
  <p></p>
  <h3>🙋‍♀️ 文章结论</h3>
  <blockquote>Tips: 实然：因果关系结论+应然：政策启示</blockquote>
  <p></p>
  <h3>📌 方法启示 </h3>
  <blockquote>Tips: 怎么设计的实验？怎么构造的函数？怎么演绎经济故事逻辑？</blockquote>
  <p></p>
  <h3>💭 思考启发</h3>
  <p></p>
```

**Reply by iampearl918** [reply]
谢谢！我一开始也想到您这个方法，因为刚接触这类软件还不熟悉代码，就同时先去试着找了下问题出在哪，昨晚看了好几个下载量多的模版帖子，隐约记得有个帖子下也有这个问题。刚刚去找了一通，发现了是导入的方式不对。
因为我之前看的软件使用说明和模版帖子，导入模版都是用的“从剪切板导入”，但这里的要自己去新建、然后粘贴模版内容。
感谢！接下来学习怎么写代码建模版～

**Comment by hzp2333** [comment]
# 关于一些常见报错

- 可能报错情况1：**_Template New Template: 1714978248098 Error: ReferenceError: topItem is not defined_**
>因为模板名称得是[Item]论文笔记的形式，问就是zotero6流传下来的底层代码。

- 可能报错情况2：**_Template Preview Error: TypeError: Zotero.ZoteroStyle.data.ztoolkit.ItemTree.fieldHooks.globalCache is undefined_**
>这里是笔记模板通过zotero style读取文件元数据，如果版本不匹配就不行。

**Comment by Carriekk** [comment]
哦买嘎！非常感谢！我这去试一下~谢谢！



---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 日期 | 2024年05月14日 22:05 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.***>***@***.***> |
| 主题 | Re: [windingwind/zotero-better-notes] 更适合看经济实证论文的笔记模板 (Discussion #973) |

zotero6和zotero7模板差距主要是zotero style插件读取文件元数据。因此理论上只需要修改zotero style读取部分。
你可以试试看这个。


  <h1 style="color:#193c47; background-color:#eef9fd; padding:8px;">
  ${(() => {
    const titleTranslation = topItem.getField("titleTranslation");
    if (titleTranslation) {
      return `(${topItem.getField("date")}) ${topItem.getField("title")} (${titleTranslation})`;
    } else {
      return `(${topItem.getField("date")}) ${topItem.getField("title")}`;
    }
  })()}
</h1>
  
  
  <table>
  
  <tr>
  <td style="color:#193c47; background-color:#dbeedd; padding:8px;">
  <b>作者:</b> ${topItem.getCreators().map((v) => v.firstName + " " + v.lastName).join("; ")}
</td>
  </tr>
  
  
  <tr>
  <td style="color:#193c47; background-color:#f3faf4; padding:8px;">
  <b style="color:#193c47;">期刊: <b style="color:#FF0000">${topItem.getField('publicationTitle')}</b></b><b style="color:#193c47;"> （发表日期: ${topItem.getField("date")}）</b>
  </td>
  </tr>
  
  
  <tr>
  <td style="color:#193c47; background-color:#dbeedd; padding:8px;">
  <b>期刊分区: </b>
  
  ${(() => {let space = " ㅤㅤ ㅤㅤ"
  return Array.prototype.map.call(Zotero.ZoteroStyle.data.ztoolkit.ItemTree.globalCache.renderCellHooks.PublicationTags(
    0,
    Zotero.ZoteroStyle.data.ztoolkit.ItemTree.fieldHooks.globalCache.getFieldHooks.PublicationTags(
      "", true, true,
      topItem, undefined)
  ).childNodes,
  e => {
    e.innerText = space + space + e.innerText + space + space;
    return e.outerHTML
  }).join(space)
  })()
  }
</td>
  </tr>
  
  
  <tr>
  <td style="color:#193c47; background-color:#f3faf4; padding:8px;">
  ${(() => {
    const attachments = Zotero.Items.get(topItem.getAttachments());
    if (attachments && attachments.length > 0) {
      return `<b>本地链接: </b><a href="zotero://open-pdf/0_${attachments[0].key}">${attachments[0].getFilename()}</a>`;
    } else {
      return `<b>本地链接: </b>`;
    }
  })()}
</td>
  </tr>
  
  
  <tr>
  <td style="color:#193c47; background-color:#dbeedd; padding:8px;">
  ${(() => {
    const doi = topItem.getField("DOI");
    if (doi) {
      return `<b>DOI: </b><a href="https://doi.org/${topItem.getField('DOI')}">${topItem.getField('DOI')}</a>`;
    } else {
      return `<b>URL: </b><a href="${topItem.getField('url')}">${topItem.getField('url')}</a>`;
    }
  })()}
</td>
  </tr>
  
  
  <tr>
  <td style="color:#193c47; background-color:#f3faf4; padding:8px;">
  ${(() => {
    const abstractTranslation = topItem.getField('abstractTranslation');
    if (abstractTranslation) {
      return `<b>摘要翻译: </b><i>${abstractTranslation}</i>`;
    } else {
      return `<b>摘要: </b><i>${topItem.getField('abstractNote')}</i>`;
    }
  })()}
</td>
  </tr>
  
  
  <tr>
  <td style="color:#193c47; background-color:#dbeedd; padding:8px;">
  <b>笔记日期: </b>${new Date().toLocaleString()}
</td>
  </tr>
  
  </table>
  
  
  <span>
  <h2 style="color:#e0ffff; background-color:#66cdaa;">📜 研究核心</h2>
  <hr />
  </span>
  <blockquote>Tips: 做了什么，解决了什么问题，创新点与不足？</blockquote>
  以xx为对象，使用了xx方法，得到了xx结论
<p></p>
  
  <span>
  <h2 style="color:#20b2aa; background-color:#afeeee;">🔁 实证部分</h2>
  <hr />
  </span>
  <p></p>
  <h3>💧 模型</h3>
  <blockquote>Tips: 理论现象路径背景和计量回归式子</blockquote>
  正文开始
<p></p>
  <h3>👩🏻‍💻 数据</h3>
  <table border="1">
  <tr>
  <td style="color:#193c47; background-color:#dbeedd; padding:8px;"><b>变量</b></td>
  <td style="color:#193c47; background-color:#dbeedd; padding:8px;"><b>含义</b></td>
  <td style="color:#193c47; background-color:#dbeedd; padding:8px;"><b>数据源</b></td>
  </tr>
  <tr>
  <td>因变量</td>
  <td> </td>
  </tr>
  <tr>
  <td>自变量</td>
  <td> </td>
  <td> </td>
  </tr>
  <tr>
  <td>控制变量</td>
  <td> </td>
  <td> </td>
  </tr>
  
  <tr>
  <td>其他变量IV等</td>
  <td></td>
  <td></td>
  </tr>
  
  <tr>
  <td>效应</td>
  <td>双向固定、聚类</td>
  <td></td>
  </tr>
  
  </table>
  
  <p></p>
  <h3>🔬 稳健性与内生性</h3>
  <p></p>
  <h3>📜 异质性</h3>
  <p></p>
  
  <h3>⚙️ 机制检验</h3>
  <p></p>
  
  <h3>💡 拓展分析</h3>
  <p></p>
  
  <span>
  <h2 style="color:#004d99; background-color:#87cefa;">🤔 总结</h2>
  <hr />
  </span>
  <blockquote>Tips: 你对哪些内容产生了疑问，你认为可以如何改进？</blockquote>
  <p></p>
  <h3>🙋‍♀️ 文章结论</h3>
  <blockquote>Tips: 实然：因果关系结论+应然：政策启示</blockquote>
  <p></p>
  <h3>📌 方法启示 </h3>
  <blockquote>Tips: 怎么设计的实验？怎么构造的函数？怎么演绎经济故事逻辑？</blockquote>
  <p></p>
  <h3>💭 思考启发</h3>
  <p></p>

**Comment by wakanda-wang** [comment]
感谢作者！zotero7版本直接使用完全没有问题 👍

**Comment by hzp2333** [comment]
详细说明
欢迎访问我的博客~
https://blog.huaxiangshan.com/zh-cn/posts/zoteronote/

**Comment by Willow0991** [comment]
请问想要直接在笔记中而不是模版中给表格添加行要怎么实现呢

**Reply by hzp2333** [reply]
笔记页面内：右键-插入-表格

**Reply by Willow0991** [reply]
谢谢！

**Comment by iampearl918** [comment]
请问better note 笔记中，如何更改引用文献的样式呢？首次插入时跳出来选择样式的窗口，之后就不再有了，想更改样式找不到地方！
备注：不是word，是zotero的note

---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 发送日期 | 2024年08月20日 08:31 |
| 收件人 | windingwind/zotero-better-notes ***@***.***> |
| 抄送人 | iampearl918 ***@***.***>,
Comment ***@***.***> |
| 主题 | Re: [windingwind/zotero-better-notes] 更适合看经济实证论文的笔记模板 (Discussion #973) |

笔记页面内：右键-插入-表格

**Reply by hzp2333** [reply]
note中的年份作者引用似乎是无法修改的，那个作用是为了方便跳转文献。

但是可以隐藏作者年份注释（右上角三点-隐藏注释）

**Reply by hzp2333** [reply]
> 请问better note 笔记中，如何更改引用文献的样式呢？首次插入时跳出来选择样式的窗口，之后就不再有了，想更改样式找不到地方！ 备注：不是word，是zotero的note
> […](#)
> ---- 回复的原邮件 ---- | 发件人 | ***@***.***> | | 发送日期 | 2024年08月20日 08:31 | | 收件人 | windingwind/zotero-better-notes ***@***.***> | | 抄送人 | iampearl918 ***@***.***>, Comment ***@***.***> | | 主题 | Re: [windingwind/zotero-better-notes] 更适合看经济实证论文的笔记模板 (Discussion #973) | 笔记页面内：右键-插入-表格 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: ***@***.***>

如果要添加其他论文引用——右键-插入-引注

**Comment by iampearl918** [comment]
你可能理解错了我的意思，我不是指添加注释时自动生成的那个引用，是自己插入的引用。

---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 发送日期 | 2024年09月02日 13:26 |
| 收件人 | windingwind/zotero-better-notes ***@***.***> |
| 抄送人 | iampearl918 ***@***.***>,
Comment ***@***.***> |
| 主题 | Re: [windingwind/zotero-better-notes] 更适合看经济实证论文的笔记模板 (Discussion #973) |

请问better note 笔记中，如何更改引用文献的样式呢？首次插入时跳出来选择样式的窗口，之后就不再有了，想更改样式找不到地方！ 备注：不是word，是zotero的note
…
---- 回复的原邮件 ---- | 发件人 | @.> | | 发送日期 | 2024年08月20日 08:31 | | 收件人 | windingwind/zotero-better-notes @.> | | 抄送人 | iampearl918 @.>, Comment @.> | | 主题 | Re: [windingwind/zotero-better-notes] 更适合看经济实证论文的笔记模板 (Discussion #973) | 笔记页面内：右键-插入-表格 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: @.***>

如果要添加其他论文引用——右键-插入-引注

**Reply by hzp2333** [reply]
试试看这个？右键，插入，引用

**Comment by iampearl918** [comment]
不行哦、这个操作和上方的快捷按钮，效果一样。没弹出来引用样式让我选

---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 发送日期 | 2024年09月02日 22:37 |
| 收件人 | windingwind/zotero-better-notes ***@***.***> |
| 抄送人 | iampearl918 ***@***.***>,
Comment ***@***.***> |
| 主题 | Re: [windingwind/zotero-better-notes] 更适合看经济实证论文的笔记模板 (Discussion #973) |

试试看这个？右键，插入，引用

**Reply by hzp2333** [reply]
是指年份作者的引用格式，改成脚注或者数字标那种吗？

如果是这种应该修改不了的。这里是markdown环境，zotero和better note没有针对md环境做引用调整的格式设置。

**Comment by iampearl918** [comment]
好像和您说的这个情况类似，我想说的就是“文中”的引用格式，作者年份的格式。在不同样式下，它们呈现出来的不同。

---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 发送日期 | 2024年09月03日 10:16 |
| 收件人 | windingwind/zotero-better-notes ***@***.***> |
| 抄送人 | iampearl918 ***@***.***>,
Comment ***@***.***> |
| 主题 | Re: [windingwind/zotero-better-notes] 更适合看经济实证论文的笔记模板 (Discussion #973) |

是指年份作者的引用格式，改成脚注或者数字标那种吗？

如果是这种应该修改不了的。这里是markdown环境，zotero和better note没有针对md环境做引用调整的格式设置。

**Comment by iampearl918** [comment]
哪怕不在同一个markdown note中调整样式，为啥其他的markdown note就默认这一个，再也改不了了呢😭

---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 发送日期 | 2024年09月03日 10:16 |
| 收件人 | windingwind/zotero-better-notes ***@***.***> |
| 抄送人 | iampearl918 ***@***.***>,
Comment ***@***.***> |
| 主题 | Re: [windingwind/zotero-better-notes] 更适合看经济实证论文的笔记模板 (Discussion #973) |

是指年份作者的引用格式，改成脚注或者数字标那种吗？

如果是这种应该修改不了的。这里是markdown环境，zotero和better note没有针对md环境做引用调整的格式设置。

**Comment by iampearl918** [comment]
甚至我刚把之前用的样式删除了，它还是原样式的格式🧐

---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 发送日期 | 2024年09月03日 10:16 |
| 收件人 | windingwind/zotero-better-notes ***@***.***> |
| 抄送人 | iampearl918 ***@***.***>,
Comment ***@***.***> |
| 主题 | Re: [windingwind/zotero-better-notes] 更适合看经济实证论文的笔记模板 (Discussion #973) |

是指年份作者的引用格式，改成脚注或者数字标那种吗？

如果是这种应该修改不了的。这里是markdown环境，zotero和better note没有针对md环境做引用调整的格式设置。

**Reply by hzp2333** [reply]
你是想改成数字标吗？markdown有自己的脚注格式，你可以手动修改，但是zotero似乎没有相关联动功能。

也可以尝试联动zotero与obsidian，这样会更方便修改格式。

参见博客：https://blog.huaxiangshan.com/zh-cn/posts/ob/

**Comment by Recardott** [comment]
首先感谢作者分享！我昨天用了您的模版没有问题，然后把Style插件升级成PRO版了，然后今天添加笔记时好像就不行了，出现Template Preview Error: SyntaxError: missing ) after argument list这个问题，不知道如何解决，向您咨询一下

**Reply by Recardott** [reply]
我解决了，可能是插件bug了，删了重下就好了
